### PR TITLE
=htt - Improve stability of ClientServerSpec

### PR DESCRIPTION
**Pull request by viktorklang**

### CODE REVIEW

**Comment by ktoso** [comment]
LGTM, thanks @viktorklang ! 
I'll use that style in https://github.com/akka/akka/pull/18823 as well then

**Comment by ktoso** [comment]
Is there a ticket tracking this `failed`?

**Comment by ktoso** [comment]
There is a number of tickets about this Spec however they seem to be talking about something else (that line is a new test I added this week, those issues are much older): https://github.com/akka/akka/search?q=ClientServerSpec&type=Issues&utf8=%E2%9C%93

File: akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala [hunk 1]
```diff
@@ -168,12 +168,9 @@ class ClientServerSpec extends WordSpec with Matchers with BeforeAndAfterAll {
...
       // waiting for the timeout to happen on the client
-      Try(Await.result(clientsResponseFuture, 2.second)).recoverWith {
-        case _: StreamTcpException ⇒ Success(System.nanoTime())
-        case other: Throwable      ⇒ Failure(other)
-      }.get
-      val diff = System.nanoTime() - serverReceivedRequestAtNanos
-      diff should be > theIdleTimeout.toNanos
+      intercept[StreamTcpException] { Await.result(clientsResponseFuture, 2.second) }
+
+      (System.nanoTime() - serverReceivedRequestAtNanos).millis should be >= theIdleTimeout
```
Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala]: Duh, that's wrong actually - passes by accident. This should be `.nanos`:

```
        (System.nanoTime() - serverReceivedRequestAtNanos).nanos should be >= theIdleTimeout
```

I've fixed that in https://github.com/akka/akka/pull/18823 .
Stopping reviewing, too jet lagged :wink:

Comment (viktorklang) [inline akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala]: In what way is it wrong? Do we have a Scheduler with nanosecond precision and accuracy? ;-)

Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala]: Uhm... It's not about precision in that sense but that nanoTime's returned value is in nanoseconds... so `(numberAsNanoseconds).nanos` makes sense `(numberAsNanoSeconds).millis` is not the right thing, no?

Comment (viktorklang) [inline akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala]: Ah, sorry, it's too late for me too. I'll fix.

Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala]: Coding when tired considered harmful :wink: (I'm guilty of that as well tho).
I have it fixed in my #18823, don't worry about it :)

Comment (viktorklang) [inline akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala]: Thanks!

**Pull request merged**